src/test/java/edu/tamu/eider/controller/EntityControllerTest.java

```diff
@@ -0,0 +1,126 @@
+package edu.tamu.eider.controller;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.document;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.get;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.head;
+import static org.springframework.restdocs.payload.PayloadDocumentation.fieldWithPath;
+import static org.springframework.restdocs.payload.PayloadDocumentation.responseFields;
+import static org.springframework.restdocs.request.RequestDocumentation.parameterWithName;
+import static org.springframework.restdocs.request.RequestDocumentation.pathParameters;
+import static org.springframework.restdocs.request.RequestDocumentation.requestParameters;
+import static org.springframework.restdocs.payload.PayloadDocumentation.subsectionWithPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+import edu.tamu.eider.app.model.Entity;
+import edu.tamu.eider.app.model.Name;
+import edu.tamu.eider.app.model.repo.EntityRepository;
+import edu.tamu.eider.app.model.repo.NameRepository;
+import edu.tamu.eider.resources.EntityTestData;
+
+public class EntityControllerTest extends EntityTestData {
+
+    private static String TEST_NAME_NAME = "Name";
+    private static String TEST_NAME_1_NOTES = "First Name";
+    private static String TEST_NAME_2_NOTES = "Second Name";
+
+    @Autowired
+    private EntityRepository entityRepo;
+
+    @Autowired
+    private NameRepository nameRepo;
+
+    @Test
+    public void testRedirectHeadToEntity() throws Exception {
+        Entity entity = entityRepo.save(TEST_ENTITY_1);
+        this.mockMvc
+            .perform(head("/entity/{id}", entity.getId().toString()))
+            .andExpect(status().isFound())
+            .andDo(document("redirect-head-to-entity",
+                pathParameters(
+                    parameterWithName("id").description("The UUID id of the Entity whose URL the response will redirect to")
+                )
+            ));
+    }
+
+    @Test
+    public void testFindByUrl() throws Exception {
+        Entity entity = entityRepo.save(TEST_ENTITY_1);
+        this.mockMvc
+            .perform(get("/entity/url")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .param("url", entity.getUrl().toString())
+            )
+            .andExpect(status().isOk())
```

Comment:
We could also expect a specific content-type. `content().contentType(MediaTypes.HAL_JSON_UTF8_VALUE)`